framework/contracts/account/manager/tests/upgrades.rs

```diff
@@ -650,6 +656,238 @@ fn create_sub_account_with_installed_module() -> AResult {
+            namespace: Some(String::from(TEST_NAMESPACE)),
+            base_asset: None,
+            install_modules: vec![],
+        },
+        GovernanceDetails::Monarchy {
+            monarch: sender.to_string(),
+        },
+        None,
+    )?;
+    deploy_modules(&chain);
+    // Add monetization
+    deployment.version_control.set_module_monetization(
+        "mock-adapter1".to_owned(),
+        Monetization::InstallFee(FixedFee::new(&coin(5, "coin1"))),
+        Namespace::new("tester").unwrap(),
+    )?;
+    deployment.version_control.set_module_monetization(
+        "mock-adapter2".to_owned(),
+        Monetization::InstallFee(FixedFee::new(&coin(5, "coin1"))),
+        Namespace::new("tester").unwrap(),
+    )?;
+    deployment.version_control.set_module_monetization(
+        "mock-app1".to_owned(),
+        Monetization::InstallFee(FixedFee::new(&coin(5, "coin2"))),
+        Namespace::new("tester").unwrap(),
+    )?;
+
+    // Check how much we need
+    let simulate_response = deployment.module_factory.simulate_install_modules(vec![
+        ModuleInfo::from_id(adapter_1::MOCK_ADAPTER_ID, V1.into()).unwrap(),
+        ModuleInfo::from_id(adapter_2::MOCK_ADAPTER_ID, V1.into()).unwrap(),
+        ModuleInfo::from_id(app_1::MOCK_APP_ID, V1.into()).unwrap(),
+    ])?;
+    assert_eq!(
+        simulate_response.required_funds,
+        vec![coin(10, "coin1"), coin(5, "coin2")]
+    );
+
+    let result = factory.create_new_account(
+        AccountDetails {
+            name: String::from("second_account"),
+            description: None,
+            link: None,
+            namespace: None,
+            base_asset: None,
+            install_modules: vec![
+                (
+                    ModuleInfo::from_id(
+                        adapter_1::MOCK_ADAPTER_ID,
+                        ModuleVersion::Version(V1.to_owned()),
+                    )?,
+                    None,
+                ),
+                (
+                    ModuleInfo::from_id(
+                        adapter_2::MOCK_ADAPTER_ID,
+                        ModuleVersion::Version(V1.to_owned()),
+                    )?,
+                    None,
+                ),
+                (
+                    ModuleInfo::from_id(app_1::MOCK_APP_ID, ModuleVersion::Version(V1.to_owned()))?,
+                    Some(to_binary(&app::InstantiateMsg {
+                        module: MockInitMsg,
+                        base: BaseInstantiateMsg {
+                            ans_host_address: deployment.ans_host.addr_str()?,
+                        },
+                    })?),
+                ),
+            ],
+        },
+        GovernanceDetails::Monarchy {
+            monarch: sender.to_string(),
+        },
+        // we attach 1 less coin1
+        Some(&[coin(9, "coin1"), coin(10, "coin2")]),
+    );
+    // Mock doesn't implement debug so we can't .unwrap_err, LOL
```

Comment:
Which mock? If there's a type that's missing the `#[derive(Debug)]` we should create an issue for it.

Reply:
AbstractAccount<Mock>